sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -4421,6 +4421,16 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val POSTGRES_PROPER_READING_OF_ARRAY_DIMENSIONALITY =
+    buildConf("spark.sql.legacy.postgres.properReadingOfArrayDimensionality")
```

Comment:
usually legacy config true means use the legacy behavior.  how about `spark.sql.legacy.postgres.getArrayDimFrom PgAttribute `.

And usually we enable bug fix by default.

Reply:
Thanks for suggestion. Changed the code now.